client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js

```diff
@@ -0,0 +1,249 @@
+					</FormLabel>
+					<FormLabel>
+						{ translate( 'Subscribe message is unchecked by default' ) }
+					</FormLabel>
+					<FormTextInput
+						name={ 'newsletter_label' }
+						onChange={ onNewslatterLabelChange }
+						value={ settings.newsletter_label }
+					/>
+				</FormFieldset>
+			</span>
+			<span className="mailchimp__dashboard-settings-preview">
+				<div>{ translate( 'PREVIEW' ) }</div>
+				<div className="mailchimp__dashboard-settings-preview-view">
+					{toggle && <FormLabel>
+							<FormCheckbox checked={ checkbox === 'check' } disabled={ true } />
+							<span>{ settings.newsletter_label }</span>
+						</FormLabel>}
+				</div>
+				<div className="mailchimp__dashboard-settings-save">
+					<Button
+						primary
+						onClick={ onSave }
+						busy={ isSubbmittingSettings }
+						disabled={ isSubbmittingSettings }>
+						{ translate( 'Save' ) }
+					</Button>
+				</div>
+			</span>
+		</div>
+	);
+} );
+
+class MailChimpDashboard extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			syncStatus: null,
+			settings: props.settings
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { translate } = nextProps;
+		if ( ( nextProps.isSubmittingNewsletterSetting === false ) && this.props.isSubmittingNewsletterSetting ) {
+			if ( nextProps.newsletterSettingsSubmitError ) {
+				nextProps.errorNotice( translate( 'There was a problem saving the email settings. Please try again.' ) );
+			} else {
+				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
+			}
+		}
+	}
+
+	onSettingsChange = ( change ) => {
+		this.setState( { settings: Object.assign( {}, this.state.settings, change ) } );
+	}
+
+	onSave = () => {
+		const { submitMailchimpNewsletterSettings: submit, siteId } = this.props;
+		const settings = this.state.settings;
+		const message = {
+			mailchimp_list: settings.mailchimp_list,
+			newsletter_label: settings.newsletter_label,
+			mailchimp_auto_subscribe: settings.mailchimp_auto_subscribe,
+			mailchimp_checkbox_defaults: settings.mailchimp_checkbox_defaults,
+			mailchimp_checkbox_action: settings.mailchimp_checkbox_action,
+		};
+		submit( siteId, message );
+	}
+
+	render() {
+		const { siteId, syncStatusData, translate } = this.props;
+		const slogan = translate( 'Allow customers to subscribe to your MailChimp email list' );
+
+		return (
+			<div>
+				<QueryMailChimpSyncStatus siteId={ siteId } />
+				<Card className="mailchimp__dashboard" >
+					<div className="mailchimp__dashboard-first-section" >
+						<span className="mailchimp__dashboard-title-and-slogan">
+							<div className="mailchimp__dashboard-title">Mailchimp</div>
+							<div>{ slogan }</div>
+						</span>
+						<span className="mailchimp__dashboard-sync-status" >
+							<SyncTab
+								siteId={ siteId }
+								syncState={ syncStatusData }
+								resync={ this.props.requestResync } />
```

Comment:
could add `requestResync` to the props assignment on line 196. Few more variables you could do the same with below.

Reply:
True but it creates name clashes with imports. What would be the best strategy here. Have a different name in connect or extract to a different name like { requestResync: reqResync } = ...
That is why I have originally left it this way.

Reply:
When I run into naming clashes like that, I just continue using `this.props.whatever` like you have.

Reply:
Most of the things tackle. Ready for more bashing :)